finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -132,8 +139,8 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
                 if (closed.get) Future.Unit
                 else transport.read().transform(loop)
 
-              case StreamClosed | StreamRemoteReset =>
-                // The stream has been closed and should know better than
+              case StreamRemoteReset =>
+                // The stream has been reset and should know better than
```

Comment:
Doesn't this make the match non-exhaustive?  Should we remove the StreamRemoteClosed state?